### PR TITLE
Vulkan : Image cleanup

### PR DESCRIPTION
**Pull request by apjagdale**
Moving creation of readbackbuffer from vk_image_base to vk_render_texture.
Fixing binding locations of textures starting it from 5 instead of 4, since shadowmap is at 4.

GearVRF DCO signed off by: Abhijit Jagdale a.jagdale@samsung.com

### CODE REVIEW

File: GVRf/Framework/framework/src/main/res/raw/phong_surface_layertex.fsh [hunk 1]
```diff
@@ -27,36 +27,36 @@ layout(location = 8) in vec2 lightmap_coord;
...
 
-layout(set = 0, binding = 5) uniform sampler2D ambientTexture;
-layout(set = 0, binding = 4) uniform sampler2D diffuseTexture;
-layout(set = 0, binding = 6) uniform sampler2D specularTexture;
-layout(set = 0, binding = 7) uniform sampler2D opacityTexture;
-layout(set = 0, binding = 8) uniform sampler2D lightmapTexture;
-layout(set = 0, binding = 10) uniform sampler2D emissiveTexture;
-layout(set = 0, binding = 9) uniform sampler2D normalTexture;
+layout(set = 0, binding = 6) uniform sampler2D ambientTexture;
+layout(set = 0, binding = 5) uniform sampler2D diffuseTexture;
```
Comment (liaxim) [inline GVRf/Framework/framework/src/main/res/raw/phong_surface_layertex.fsh]: Can't we have all these bindings #define-ed somewhere in one place?

File: GVRf/Framework/framework/src/main/jni/vulkan/vk_imagebase.cpp [hunk 2]
```diff
@@ -102,127 +65,81 @@ int getComponentsNumber(VkFormat format){
...
+        );
         GVR_VK_CHECK(!ret);
-
-        ret = vkBindBufferMemory(device, hostBuffer, host_memory, 0);
+    }
+    else {
+        // View for whole array
+        ret = vkCreateImageView(
+                device,
+                gvr::ImageViewCreateInfo(imageHandle, VK_IMAGE_VIEW_TYPE_2D_ARRAY,
```
Comment (roshanch) [inline GVRf/Framework/framework/src/main/jni/vulkan/vk_imagebase.cpp]: why cann't we use imageType here? please check whether  it is properly set.

Comment (roshanch) [reply GVRf/Framework/framework/src/main/jni/vulkan/vk_imagebase.cpp]: can we get rid of ifs

File: GVRf/Framework/framework/src/main/res/raw/phong_surface_layertex.fsh (hunk 1 above)
Comment (roshanch) [reply GVRf/Framework/framework/src/main/res/raw/phong_surface_layertex.fsh]: yeah lets add all bindings into separate file with #define Vulkan tag.

**Comment by liaxim** [comment]
@apjagdale Is this it for the pull request? No further changes planned? What about @roshanch's comments?

**Comment by apjagdale** [comment]
@liaxim I did the changes specified by @roshanch . Regarding binding points, we need to come up with an approach, which can be done in a different pull request.

**Comment by liaxim** [comment]
Planning to merge on Tuesday

**Pull request merged**